compiler/src/dotty/tools/dotc/typer/ProtoTypes.scala

```diff
@@ -653,6 +654,18 @@ object ProtoTypes {
 
   final def wildApprox(tp: Type)(implicit ctx: Context): Type = wildApprox(tp, null, Set.empty)
 
+  final def representsKind(tl: HKTypeLambda)(implicit ctx: Context): Boolean = {
+    tl.paramInfos.forall {
+      case TypeBounds(lo, hi) =>
+        def isKindBound(tp: Type): Boolean =
+          Inferencing.isFullyDefined(tp, ForceDegree.none) || (tp match {
```

Comment:
It would be good to have some explanations about the logic here as a comment. In particular I did not understand why a fully defined bounds represents a kind.

Reply:
I am also worried about performance implications. `isFullyDefined` is not cheap, and `wildApprox` is a hotspot.

Reply:
I'm just finishing up with a different approach ... no need to continue reviewing this.